src/Aspire.Hosting/Extensions/ContainerResourceBuilderExtensions.cs

```diff
@@ -67,6 +68,38 @@ public static IResourceBuilder<ContainerResource> AddContainer(this IDistributed
         return builder.WithAnnotation(annotation);
     }
 
+    /// <summary>
+    /// Exposes an HTTP endpoint on a resource. This binding reference can be retrieved using <see cref="ResourceBuilderExtensions.GetEndpoint{T}(IResourceBuilder{T}, string)"/>.
+    /// The binding name will be "http" if not specified.
+    /// </summary>
+    /// <typeparam name="T">The resource type.</typeparam>
+    /// <param name="builder">The resource builder.</param>
+    /// <param name="containerPort">The container port.</param>
+    /// <param name="hostPort">The host machine port.</param>
+    /// <param name="name">The name of the binding.</param>
+    /// <param name="env">The name of the environment variable to inject.</param>
+    /// <returns>The <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<T> WithHttpEndpoint<T>(this IResourceBuilder<T> builder, int containerPort, int? hostPort = null, string? name = null, string? env = null) where T : IResource
```

Comment:
`ContainerResource`? as constraint?

Reply:
I copied this signature from https://github.com/dotnet/aspire/blob/fc0d681c0921984da03e61dcbf69ce5528a658d3/src/Aspire.Hosting/Extensions/ContainerResourceBuilderExtensions.cs#L52

It has resource instead of container resource. Do you know why?

Reply:
We don’t want the container resource constraint. This has to work for executables that deploy as a container as well (docker file scenario).

Reply:
Should we add something to the doc comments to point out that this overload is intended for resources that run or publish as containers, given that `containerPort` is required.

Reply:
Should the overload be in `ContainerResourceBuilderExtensions`? It's not specific to a container resource.

I'm guessing no.

Reply:
Sure